traffic_mirroring.go

```diff
@@ -30,36 +25,36 @@ func main() {
 	}
 
 	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
-	max_size := 10
-	packets_list := make([][]byte,max_size) 
-	current_packet := 0
+	packetsList := make(chan []byte, 22)
+	filePath := "Test"
+	go writeFile(packetsList, filePath)
 
 	for overlayPacket := range packetSource.Packets() {
 		vxlanLayer := overlayPacket.Layer(layers.LayerTypeVXLAN)
 		if vxlanLayer != nil {
 			vxlanPacket, _ := vxlanLayer.(*layers.VXLAN)
-			
-			fmt.Println(current_packet)
-			packets_list[current_packet] = vxlanPacket.LayerPayload()
-			current_packet += 1
+			packetsList <- vxlanPacket.LayerPayload()
 			fmt.Println(vxlanPacket.LayerPayload())
-			if current_packet == max_size {
-				newSlice := make([][]byte, max_size)
-				copy(newSlice, packets_list)
-				fmt.Println(current_packet)
-				current_packet = 0
-				go write_file(newSlice, f)
-
 
-			}
 		}
 	}
-	defer f.Close()
 }
 
-func write_file(packets_list [][]byte, f *os.File) {
-	for _, value := range packets_list {
-		fmt.Fprintln(f, value)  
+func writeFile(packetsList chan []byte, filePath string) {
+	numberPackets := 20
+	f, err := os.Create(filePath)
```

Comment:
To facilitate your unit tests for this function (to not deal with complex file operation, such as creation/deletion). Please, create this file in the main code, and encapsulate its object inside the interface `io.Writer`. You can see an example [here](https://github.com/vtex/reliability-research/blob/master/load_test/inputprocessor.go#L9), and the unit test for the example can be accessed [here](https://github.com/vtex/reliability-research/blob/master/load_test/inputprocessor_test.go#L10).